src/widgets/search-box/__tests__/search-box-test.ts

```diff
@@ -1,10 +1,14 @@
 import { render } from 'preact';
 import searchBox from '../search-box';
+import algoliaSearchHelper, { AlgoliaSearchHelper } from 'algoliasearch-helper';
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
 import {
   createInitOptions,
   createRenderOptions,
 } from '../../../../test/mock/createWidget';
 
+const mockedRender = render as jest.Mock;
```

Comment:
the other tests are like this: 

https://github.com/algolia/instantsearch.js/blob/ceb48c2d88a9f237ab1be8520b5725ab19433d43/src/widgets/breadcrumb/__tests__/breadcrumb-test.ts#L1-L16

Reply:
Done https://github.com/algolia/instantsearch.js/pull/4710/commits/9d467c7c3b9dc048630dcf1111d7b51c48263fc6

Has some issues with the VNode afterward (the `castToJestMock` being more precise from what I had done).

I had to do some kind of hacky casting to get the proper types in the tests 
https://github.com/algolia/instantsearch.js/pull/4710/files#diff-eb04715aca8f31f90866aef1d48d5f87f77c33223c0b7bfff89e908626533591R92

I saw that in other tests you added a `// @ts-expect-error` instead of trying to cast, I can do the same thing if you prefer this over the hacky casting :wink: